k8s/showcase/vineyard-mars-pytorch/process-data/resources/cluster-role.yaml

```diff
@@ -0,0 +1,38 @@
+# Copyright 2020-2023 Alibaba Group Holding Limited.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: process-data-role
+  labels:
+    app.kubernetes.io/instance: process-data
+    # Add these permissions to the "admin" and "edit" default roles.
+    rbac.authorization.k8s.io/aggregate-to-admin: "true"
+    rbac.authorization.k8s.io/aggregate-to-edit: "true"
+rules:
+  - apiGroups: [""]
+    resources: ["pods", "pods/exec", "pods/log", "endpoints", "services","namespaces","nodes"]
+    verbs: ["get", "patch", "delete", "create", "watch", "list"]
+  - apiGroups: ["rbac.authorization.k8s.io"]
+    resources: ["roles", "rolebindings"]
+    verbs: ["patch", "get", "create", "delete"]
+  - apiGroups: ["rbac.authorization.k8s.io"]
+    resources: ["clusterroles"]
+    verbs: ["bind"]
+    resourceNames: ["admin","edit","view"]
+  - apiGroups: ["apps"]
+    resources: ["deployments"]
```

Comment:
Should it be part of vineyardd deployment rather than job-specific configuration?

Reply:
The role is job-specific configuration, as Mars will create their own resources(deployments、service, etc) using a black-box way.